Update CHANGELOG for 0.23.0 release (#647)
temporalio/sdk-typescript: CHANGELOG.md

Breaking changes marked with a :boom:

## [0.23.0] - 2022-05-12

### Bug Fixes

- [`workflow`] Use "correct" default retry policy for local activities ([#630](https://github.com/temporalio/sdk-typescript/pull/630))
- [`activity`] Fix Activity resolved as cancelled in unsupported situations ([#640](https://github.com/temporalio/sdk-typescript/pull/640))
- [`workflow`] Propagate scope cancellation while waiting on ExternalWorkflowHandle cancellation ([#633](https://github.com/temporalio/sdk-typescript/pull/633))

> - Improve Payload Converter logic ([#558](https://github.com/temporalio/sdk-typescript/pull/558))
>   - Return undefined from `PayloadConverter.toPayload(value)` when `value` is not of a supported type.
>   - The SDK now throws when it receives `undefined` from `toPayload`

Reviewer: ```suggestion
- :boom: Improve Payload Converter logic ([#558](https://github.com/temporalio/sdk-typescript/pull/558))

  BREAKING CHANGE:
  - `PayloadConverter.toPayload(value)` now returns `undefined` when `value` is not of a supported type.
  - The SDK now throws when it receives `undefined` from `toPayload`
```

### Features

- :boom: Allow for multiple `DataConverter.payloadCodecs` ([#643](https://github.com/temporalio/sdk-typescript/pull/643))

>   `DataConverter.payloadCodec` was changed to plural:

Reviewer: ```suggestion
  BREAKING CHANGE: `DataConverter.payloadCodec` was changed to plural:
```

  ```ts
  export interface DataConverter {
    ...
    payloadCodecs?: PayloadCodec[];
  }
  ```

- [`worker`] Support HTTP headers in `NativeConnection` ([#644](https://github.com/temporalio/sdk-typescript/pull/644))
- :boom: [`worker`] Restructure `TelemetryOptions` ([#646](https://github.com/temporalio/sdk-typescript/pull/646))

>   - Also support passing HTTP headers to an OpenTelemetry Collector

Reviewer: ```suggestion

  BREAKING CHANGE
  - Also support passing HTTP headers to an OpenTelemetry Collector
```
